src/main/resources/public/jsx/geo-dash-widget-editor.js

```diff
@@ -783,33 +771,22 @@ class BasicLayout extends React.PureComponent{
             FormReady: true
         });
     };
-    checkDatesDual() {
-        let ed = new Date(this.state.endDateDual);
-        let sd = new Date(this.state.startDateDual);
+    checkDates(isDual) {
+        const ed = isDual != null? new Date(this.state.endDateDual) : new Date(this.state.endDate);
+        const sd = isDual != null? new Date(this.state.startDateDual) : new Date(this.state.startDate);
+        let isFormReady = null;
         if(! this.state.dualLayer) {
-            if (ed > sd && this.state.FormReady !== true) {
-                this.setState({FormReady: true});
-            }
-            else if (ed < sd) {
-                if (this.state.FormReady === true) {
-                    this.setState({FormReady: false});
-                }
-            }
+            isFormReady = ed > sd && this.state.FormReady !== true ? true : ed < sd &&  this.state.FormReady === true? false : null;
```

Comment:
this.state.FormReady is not initialized in the state declaration, and should be camel case.

Reply:
It's initialized in the constructor, i'll change to camel